starter-code/public/javascripts/financial-data.js

```diff
@@ -0,0 +1,68 @@
+axios.get('http://api.coindesk.com/v1/bpi/historical/close.json')
+  .then(response => {
+    printTheChart(response.data.bpi);
+  })
+  .catch(error => {
+    console.log(error);
+  });
+
+
+const printTheChart = (bitCoinData => {
+  const bitCoinLabels = Object.entries(bitCoinData).map(element => element[0]);
+  const bitCoinPrice = Object.entries(bitCoinData).map(element => element[1]);
+
+  console.log("cijene su:", bitCoinPrice);
+  const max = Math.max(...bitCoinPrice);
+  const min = Math.min(...bitCoinPrice);
+  const currencyHolder = document.getElementById("currency");
+  const currency = currencyHolder.options[currencyHolder.selectedIndex].value;
+  document.getElementById("max-price").innerHTML=`${max} ${currency}`;
+  document.getElementById("min-price").innerHTML=`${min} ${currency}`;
+
+
+
+  const ctx = document.getElementById('myChart').getContext('2d');
+  const chart = new Chart(ctx, {
+    type: 'line',
+    data: {
+      labels: bitCoinLabels,
+      datasets: [{
+        label: "BitCoin Price Index",
+        backgroundColor: 'rgb(255, 99, 132)',
+        borderColor: 'rgb(255, 99, 132)',
+        data: bitCoinPrice,
+      }]
+    }
+  });
+});
+
+document.getElementById("submit-button").addEventListener('click', function () {
+
+  commonFunction();
+
+})
+
+
+document.getElementById("currency").addEventListener('change', function () {
+
+  commonFunction();
```

Comment:
I think it would be nice to auto-fill the values of the date inputs when you retrieve the first batch of dat , or at least perform check for the date values before you fire the request, adding the parameters only if they have a value attached.  If you have an on change event listener this means that the request will be sent with empty parameters if you change the currency.